### ISSUE TITLE
Separate CLI with DataPipelineDef

### ISSUE THREAD

**Issue by realstraw** [krux/hyperion#271]
Make `DataPipelineDef` not mixing in `HyperionCli`, so application need to mixing in `HyperionCli` explicitly. This is required that sometimes traits need to be created, and having a companion object with Cli. The current implementation will trigger error:

```
ThePipeline has a main method with parameter type Array[String], but com.krux.ThePipeline will not be a runnable program.
```

**Comment by sethyates** [comment]
Big :+1:

**Comment by realstraw** [comment]
fixed by dcbe808
